python/pyarrow/array.pxi

```diff
@@ -994,6 +994,27 @@ cdef class Array(_PandasConvertible):
                 size += buf.size
         return size
 
+    def get_refererenced_buffer_size(self):
+        cdef:
+            shared_ptr[CArray] shd_ptr_c_array
+            CArray *c_array
+            CResult[int64_t] c_res_buffer
+        
+        shd_ptr_c_array = pyarrow_unwrap_array(self)
+        c_array = shd_ptr_c_array.get()
+        c_res_buffer = ReferencedBufferSize(c_array[0])
+        size = GetResultValue(c_res_buffer)
+        return size
+
+    @property
+    def get_buffer_size(self):
```

Comment:
I believe the goal was for `nbytes` to use `get_referenced_buffer_size` and for the old `nbytes` behavior to become a new method.

Note: this was not what was said in the JIRA, but based on comments, I think it is what we wanted.  I have updated the description in the JIRA to be more clear on this.

Reply:
I see. I will update the PR 👍